modules/multiplayer/scene_cache_interface.cpp

```diff
@@ -204,6 +207,29 @@ Error SceneCacheInterface::_send_confirm_path(Node *p_node, NodeCache &p_cache,
 	return err;
 }
 
+ObjectID SceneCacheInterface::resolve_object_id(int p_id) const {
+  if (assigned_ids.has(p_id)) {
+    return assigned_ids.get(p_id);
+  }
+
+  return ObjectID{ (uint64_t)0 };
+}
+
+int SceneCacheInterface::get_network_id(const Object* p_object, int p_peer_id) const {
+  ERR_FAIL_NULL_V(p_object, 0);
+
+  ObjectID oid = p_object->get_instance_id();
+  const NodeCache* nc = nodes_cache.getptr(oid);
+  ERR_FAIL_NULL_V_MSG(nc, 0, vformat("No node cache found for ObjectID %d!", (uint64_t)oid));
+
+  // if (!nc->confirmed_peers.has(p_peer_id) || !nc->confirmed_peers.get(p_peer_id) || !nc->recv_ids.has(p_peer_id)) {
+  //   // Peer hasn't confirmed ID
+  //   return -1;
+  // }
+
+  return nc->cache_id;
```

Comment:
As far as I can tell, this is the ID that's ultimately used in the RPC packet, but correct me if I'm wrong.